pkg/image/containersimageregistry/registry.go

```diff
@@ -0,0 +1,290 @@
+package containersimageregistry
+
+import (
+	"archive/tar"
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/containerd/containerd/archive"
+	"github.com/containers/common/pkg/auth"
+	"github.com/containers/image/v5/copy"
+	"github.com/containers/image/v5/docker"
+	"github.com/containers/image/v5/docker/reference"
+	"github.com/containers/image/v5/image"
+	"github.com/containers/image/v5/oci/layout"
+	"github.com/containers/image/v5/pkg/compression"
+	"github.com/containers/image/v5/pkg/docker/config"
+	"github.com/containers/image/v5/signature"
+	"github.com/containers/image/v5/types"
+	dockerconfig "github.com/docker/cli/cli/config"
+	"oras.land/oras-go/v2/content/oci"
+
+	orimage "github.com/operator-framework/operator-registry/pkg/image"
+)
+
+var _ orimage.Registry = (*Registry)(nil)
+
+type Registry struct {
+	sourceCtx *types.SystemContext
+	cache     *cacheConfig
+}
+
+var DefaultSystemContext = &types.SystemContext{OSChoice: "linux"}
+
+func New(sourceCtx *types.SystemContext, opts ...Option) (orimage.Registry, error) {
+	reg := &Registry{
+		sourceCtx: sourceCtx,
+	}
+
```

Comment:
```suggestion
if sourceCtx == nil {
   reg.sourceCtx = &types.SystemContext{}
```
would avoid checks at 117 & 157.

Reply:
So technically, it is valid to send a `nil` SystemContext all the way into the `container/image` library calls. So the reason I have things setup this way is to only actually make a `types.SystemContext` when we actually _need_ to set a field. Otherwise, we roll with `nil` all the way through.

But I don't think there is a semantic difference between `&types.SystemContext{}` and `nil`, so I can get on board with this suggestion!